### PR TITLE
Support boolean JSON schemas

### PR DESCRIPTION
**Pull request by hudson-ai**
We were previously supporting `True`/`False` schemas only when nested in certain places like `items` and `additionalProperties`. This expands our coverage to handle top-level boolean schemas as well.

Note that we will now raise a `ValueError` for `False` schemas, simply because there is nothing we can generate in this case. Alternatively, we return a null grammar... But this may have unforseen consequences.

Added some minimal tests, as we already have decent coverage for the nested cases in the `TestEmptySchema` class. 

Note: developed alongside #1014, which made it much easier to tell whether I had full coverage of every type we accept :)

### CODE REVIEW

**Comment by hudson-ai** [comment]
Will merge if there are no objections @Harsha-Nori & @riedgar-ms

File: guidance/library/_json.py [hunk 1]
```diff
@@ -329,9 +326,14 @@ def _gen_json_any(lm):
...
 def _gen_json(
     lm,
-    json_schema: Mapping[str, Any],
+    json_schema: JSONSchema,
     definitions: Mapping[str, Callable[[], GrammarFunction]],
 ):
+    if json_schema is True:
+        json_schema = {}
+    elif json_schema is False:
+        raise ValueError("No valid JSON can be generated from a schema of `False`")
```
Comment (Harsha-Nori) [inline guidance/library/_json.py:335]: thoughts on raising error vs. returning empty string?

Comment (Harsha-Nori) [reply guidance/library/_json.py:335]: Intuitively I feel like warning + empty string is more "correct", but I'm easily convinced otherwise.

Comment (hudson-ai) [reply guidance/library/_json.py:335]: The empty string isn't valid against the `False` schema. I think we need to raise an exception if someone wants to generate against a schema with no valid instances. Another "bad" schema by this standard is
```
{
  "type": "object",
  "properties": {
    "name": false
  },
  "required": ["name"]
  }
  ```
  
 Note that raising an exception upon encountering a `false` subschema isn't actually correct though. See the following example:
 ```
{
  "type": "object",
  "properties": {
    "name": false
  },
  "required": []
  }
```
This should validate against **any object that does not have the key `"name"`**. Will not support for now -- I think I'll open an issue against this just to track it...

**Pull request merged**